agent/taskresource/grpcclient/credentialsfetcherclient.go

```diff
@@ -0,0 +1,122 @@
+	}
+	return conn, nil
+
+}
+
+// GetSocketAddress() returns the credentials-fetcher socket dir
+func GetSocketAddress() (string, error) {
+	credentialsfetcherHostDir := os.Getenv("CREDENTIALS_FETCHER_HOST_DIR")
+
+	_, err := os.Stat(credentialsfetcherHostDir)
+	if err != nil {
+		return "", err
+	}
+	return "unix:" + credentialsfetcherHostDir, nil
+}
+
+func NewCredentialsFetcherClient(conn *grpc.ClientConn, timeout time.Duration) CredentialsFetcherClient {
+	return CredentialsFetcherClient{
+		conn:    conn,
+		timeout: timeout,
+	}
+}
+
+// Credentials fetcher is a daemon running on the host which supports gMSA on linux
+type CredentialsFetcherResponse struct {
+	//lease id is a unique identifier associated with the kerberos tickets created for a container
+	LeaseID string
+	//path to the kerberos tickets created for the service accounts
+	KerberosTicketPaths []string
+}
+
+// AddKerberosLease() invokes credentials fetcher daemon running on the host
+// to create kerberos tickets associated with gMSA accounts
+func (c CredentialsFetcherClient) AddKerberosLease(ctx context.Context, credentialspecs []string) (CredentialsFetcherResponse, error) {
+	if len(credentialspecs) == 0 {
+		seelog.Error("credentialspecs request should not be empty")
+		return CredentialsFetcherResponse{}, nil
+	}
+
+	defer c.conn.Close()
+	client := pb.NewCredentialsFetcherServiceClient(c.conn)
+
+	request := &pb.CreateKerberosLeaseRequest{CredspecContents: credentialspecs}
+
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(c.timeout))
+	defer cancel()
+
+	response, err := client.AddKerberosLease(ctx, request)
+	if err != nil {
+		seelog.Error("could not create kerberos tickets: %v", err)
+		return CredentialsFetcherResponse{}, err
+	}
+	seelog.Infof("created kerberos tickets and associated with lease_id: %s", response.GetLeaseId())
+
+	credentialsFetcherResponse := CredentialsFetcherResponse{
+		LeaseID:             response.GetLeaseId(),
+		KerberosTicketPaths: response.GetCreatedKerberosFilePaths(),
+	}
+
+	return credentialsFetcherResponse, nil
+}
+
+// DeleteKerberosLease() invokes credentials fetcher daemon running on the host
+// to delete kerberos tickets of gMSA accounts associated with the leaseid
+func (c CredentialsFetcherClient) DeleteKerberosLease(ctx context.Context, leaseid string) (CredentialsFetcherResponse, error) {
+	if len(leaseid) == 0 {
+		seelog.Error("invalid leaseid provided")
+		return CredentialsFetcherResponse{}, nil
+	}
+
+	defer c.conn.Close()
+	client := pb.NewCredentialsFetcherServiceClient(c.conn)
+
+	request := &pb.DeleteKerberosLeaseRequest{LeaseId: leaseid}
+
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(c.timeout))
+	defer cancel()
+
+	response, err := client.DeleteKerberosLease(ctx, request)
+	if err != nil {
+		seelog.Error("could not delete kerberos tickets: %v", err)
+		return CredentialsFetcherResponse{}, err
+	}
+	seelog.Infof("deleted kerberos associated with lease_id: %s", response.GetLeaseId())
```

Comment:
please use a consistent `leaseID` case across this and the corresponding test file below

Reply:
updated leaseID field